### PR TITLE
Bug sweep

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Prevent errors during REST calls from being swallowed
- Fix defunct tests
- Restore correct notification message when edit fails
- Markdown: Don't render title in anchors when title is null.
- Display correct login error messages when login fails in admin
- Fixed bug where server fails to response when a login is performed using an email with mixed case.
- Prevent a user from resubmitting the same username when editing its username on the server and client side

### CODE REVIEW

**Review by kgardnr — APPROVED** [review]
Confirming this fixes all the bugs listed there! I'm good to merge once @okbel approves :)

File: client/coral-admin/src/actions/auth.js
Comment (okbel) [inline client/coral-admin/src/actions/auth.js:59]: oh god. Thank you! 😱

**Review by okbel — APPROVED** [review]

**Pull request merged**
